src/screens/TopicPageScreen/elements/NavHeader.js

```diff
@@ -107,29 +118,86 @@ const NavHeader = (props) => {
+
   return (
-    <Animated.View style={styles.container(animatedHeight)}>
+    <View style={{}}>
       <StatusBar barStyle="dark-content" />
       <View
-        source={{uri: coverPath}}
-        style={[styles.navContainer(isHeaderHide, topPosition), heightWithCoverImage()]}
+        style={[
+          styles.navContainer(isHeaderHide),
+          {
+            height: dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT_COVER
+          }
+        ]}
         imageStyle={{opacity: isHeaderHide ? 0 : 1}}>
-        <Animated.View style={styles.headerImage(opacityHeaderAnimation)}>
-          <FastImage source={{uri: coverPath}} style={heightWithCoverImage()} />
-        </Animated.View>
-        <TouchableOpacity onPress={() => backScreen()} style={styles.backbutton}>
-          {coverPath && !isHeaderHide ? (
-            <MemoIcArrowBackCircle width={normalize(32)} height={normalize(32)} />
-          ) : (
-            <MemoIcArrowBack width={normalize(24)} height={normalize(24)} />
-          )}
-        </TouchableOpacity>
-        <View style={styles.containerAction}>
+        {isLoading ? (
+          <Shimmer width={displayWidth} height={dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT_COVER} />
+        ) : (
+          <>
+            <View
+              style={{
+                width: '100%',
+                height: dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT_COVER
+              }}>
+              <FastImage
+                source={{uri: coverPath}}
+                style={{
+                  height: dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT_COVER
+                }}
+              />
+              <AnimatedBlurView
+                style={[
+                  StyleSheet.absoluteFillObject,
+                  {
+                    opacity: opacityImage
+                  }
+                ]}
+                blurType="light"
+              />
+            </View>
+          </>
+        )}
+
+        <View
+          style={{
+            flexDirection: 'row',
+            alignItems: 'center',
+            paddingTop: insets.top,
+            zIndex: 2,
+            position: 'absolute'
+          }}>
+          <TouchableOpacity onPress={() => backScreen()} style={styles.backbutton}>
+            {coverPath && !isHeaderHide ? (
+              <MemoIcArrowBackCircle width={normalize(32)} height={normalize(32)} />
+            ) : (
+              <MemoIcArrowBack width={normalize(24)} height={normalize(24)} />
+            )}
+          </TouchableOpacity>
+          <Animated.View
+            style={{
+              opacity: opacityImage
+            }}>
+            <TopicDomainHeader {...props} />
+          </Animated.View>
+        </View>
+
+        <View
+          style={
+            (styles.containerAction,
+            {
+              paddingTop: insets.top,
+              paddingRight: dimen.normalizeDimen(20),
+              zIndex: 2,
+              position: 'absolute',
+              right: 0
+            })
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are several changes here related to how the header and its elements are rendered. The introduction of the `Shimmer` component for loading states is a good practice for improving user experience. However, I noticed that you're using absolute positioning (`position: 'absolute'`) on line 173 and 197. While this can work in some cases, it can also lead to layout issues especially when dealing with different screen sizes and orientations. Consider using flexbox or other layout strategies to position your elements.



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
insets.top make sure it's looks same from the top position